lib/model/SignUpStatus.dart

```diff
@@ -0,0 +1,85 @@
+import 'package:openfoodfacts/model/Status.dart';
+
+class SignUpStatus extends Status {
+  final SignUpStatusError? statusError;
+
+  SignUpStatus._({
+    dynamic status,
+    String? statusVerbose,
+    String? body,
+    String? error,
+    int? imageId,
+    this.statusError,
+  }) : super(
+          status: status,
+          statusVerbose: statusVerbose,
+          body: body,
+          error: error,
+          imageId: imageId,
+        );
+
+  // Sentences are available in
+  // [https://github.com/openfoodfacts/openfoodfacts-translations/blob/main/openfoodfacts-web/openfoodfacts-web-fr.po]
+  factory SignUpStatus(Status status) {
+    if (status.body == null) {
+      return SignUpStatus._(
+        status: 500,
+        statusError: SignUpStatusError.UNKNOWN,
+        error:
+            'No response, open an issue here: https://github.com/openfoodfacts/openfoodfacts-dart/issues/new',
+      );
+    } else if (status.body!.contains('loggedin')) {
+      return SignUpStatus._(
+        status: 201,
+      );
+    } else if (status.body!
+        .contains('This username already exists, please choose another.')) {
+      return SignUpStatus._(
+        status: 400,
+        statusError: SignUpStatusError.USERNAME_ALREADY_USED,
+        error: 'This username already exists, please choose another.',
+      );
+    } else if (status.body!.contains('The e-mail address is already used.')) {
+      return SignUpStatus._(
+        status: 400,
+        statusError: SignUpStatusError.EMAIL_ALREADY_USED,
+        error:
+            'The e-mail address is already used by another user. Maybe you already have an account? You can reset the password of your other account.',
+      );
+    } else if (status.body!
+            .contains('The password needs to be a least 6 characters long.') ||
+        status.body!.contains(
+            'The password and confirmation password are different.')) {
+      return SignUpStatus._(
+        status: 400,
+        statusError: SignUpStatusError.INVALID_PASSWORD,
+        error:
+            'The e-mail address is already used by another user. Maybe you already have an account? You can reset the password of your other account.',
+      );
+    } else if (status.body!.contains(
+        'The user name must contain only unaccented letters, digits and dashes.')) {
+      return SignUpStatus._(
+        status: 400,
+        statusError: SignUpStatusError.INVALID_USERNAME,
+        error:
+            'The e-mail address is already used by another user. Maybe you already have an account? You can reset the password of your other account.',
+      );
+    } else {
+      return SignUpStatus._(
+        status: 400,
+        statusError: SignUpStatusError.UNKNOWN,
+        error: 'Unrecognized request error',
+      );
+    }
+  }
+}
+
+enum SignUpStatusError {
+  EMPTY_EMAIL,
+  EMAIL_ALREADY_USED,
+  EMPTY_USERNAME,
```

Comment:
Never used, is it?

Reply:
Actually it's possible to raise this kind of error.
But I don't know how to fix it without breaking everything.

Let's take the empty email address.
- If I check for this before sending the request -> not retro compatible
- If I check for this in the response, I will only have "invalid email"
In that case, I have to check for the error in the body + check with the data sent to the server

For you, what's the right way to implement this?

Reply:
I understand your questioning.
To me the method is supposed to return what the server said. The server never says that the email is empty: that's a check you have to do before.

This is what I imagine, assuming that you do this for good localizations:
* in the app
  * check if the email is empty, and localize "hey, the email is empty"
  * check if the username is empty, and localize "hey, the username is empty"
  * check if the password is empty, and localize "hey, the password is empty"
  * check if both passwords are equal, and localize "hey, both passwords must be identical"
* so now we're at the maximum of what we could do in the app; let's call the server
* the server now returns a `SignUpStatus` and we should localize each `enum` values

I don't see why this would not be retro-compatible - developers that used `Status` still get a `Status` and we just add data that they won't use.

You might have coded that directly in Smoothie in a first approach, just checking that everything was fine, with the `enum` and a `static` method that returns the `enum` value from the `Status`.
And once it's stabilized, move it to off-dart.